### PR TITLE
[spirv] Handle nested structs for [[vk::offset]]

### PR DESCRIPTION
**Pull request by antiagainst**
Also update the SPIR-V doc about [[vk::offset]]

### CODE REVIEW

File: docs/SPIR-V.rst [hunk 1]
```diff
@@ -713,6 +716,21 @@ We will have the following offsets for each member:
 ``g_float2_2``   160    160    176    112    88    128
 ============== ====== ====== ====== ====== ====== ======
 
+If the above layout rules do not satisfy your needs and you want to manually
+control the layout of struct members, you can use either
+
+* The native HLSL ``:packoffset()`` attribute: only available for cbuffers; or
+* The Vulkan-specific `[[vk::offset()]]`` attribute: applies to all resources.
```
Comment (ehsannas) [inline docs/SPIR-V.rst]: missing a single \` at the beginning of `[[vk::offset()]]``

File: tools/clang/lib/SPIRV/TypeTranslator.cpp [hunk 2]
```diff
@@ -1832,11 +1833,20 @@ TypeTranslator::getAlignmentAndSize(QualType type, LayoutRule rule,
...
-      else
+      } else {
         structSize = roundToPow2(structSize, memberAlignment);
+      }
+
+      // Reset the current offset to the one specified in the source code
+      // if exists. It's debatable whether we should do sanity check here.
+      // If the developers want manually control the layout, we leave
+      // everything to them.
+      if (const auto *offsetAttr = field->getAttr<VKOffsetAttr>()) {
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/TypeTranslator.cpp]: We could do this check before the if-else clause above to save time? i.e. 
```
if (const auto *offsetAttr = field->getAttr<VKOffsetAttr>())  {
..
} else if (rule == ... || rule == ... || rule == ...) {
..
} else {
..
}
```

Probably not a big difference though...

File: docs/SPIR-V.rst (hunk 1 above)
Comment (antiagainst) [reply docs/SPIR-V.rst]: Good catch! Done!

**Comment by antiagainst** [comment]
Only doc fix. Bots happy previously.

**Pull request merged**